lib-py/casmutils/xtal/symmetry.py

```diff
@@ -33,7 +38,49 @@ def make_factor_group(structure, tol):
     list(cu.sym.CartOp)
 
     """
-    return [CartOp(op) for op in _xtal.make_factor_group(structure._pybind_value,tol)]
+    return [CartOp(op) for op in _xtal.make_factor_group(structure._pybind_value, tol)]
+
+
+def make_perm_rep_factor_group(structure, tol):
+    """Calculates all the symmetry operations in their permutation
+    representation that map the given structure onto itself
+
+    Example: Consider a structure with 3 sites, and a symmetry
+    operation that permutates site 1 with site 2, and site 3 remains
+    the same. Then this function returns a 3x1 numpy array [1,0,2]
+
+    Parameters
+    ----------
+    structure : Structure
+    tol : float
+
+    Returns
+    -------
+    list(np.array)
+
+    """
+    factor_group = make_factor_group(structure, tol)
+    basis_sites = structure.basis_sites()
+    perm_rep_factor_group = []
+
+    for sym_op in factor_group:
+        perm_rep_sym_op = np.zeros(shape=len(basis_sites))
+
+        transformed_sites = [Site(Coordinate(np.matmul(sym_op.matrix, site.cart(
```

Comment:
You seem to be applying symmetry to a site. This is something that's going to show up a lot of times. The routine that generates a permutation representation of the factor group shouldn't also be the one that implements how sites transform under symmetry.